src/pages/GameSelectionScreen.vue

```diff
@@ -389,41 +390,20 @@ export default class GameSelectionScreen extends Vue {
         await this.$store.dispatch('checkMigrations');
         this.runningMigration = false;
 
-        ManagerSettings.getSingleton(GameManager.unsetGame()).then(settings => {
-            const lastSelectedGame = settings.getContext().global.lastSelectedGame;
-            const savedViewMode = settings.getContext().global.gameSelectionViewMode;
-            switch (savedViewMode) {
-                case "List": this.viewMode = GameSelectionViewMode.LIST; break;
-                case "Card":
-                case undefined:
-                    this.viewMode = GameSelectionViewMode.CARD;
-                    break;
-            }
-            if (lastSelectedGame !== null) {
-                const game = GameManager.gameList.find(value => value.internalFolderName === lastSelectedGame);
-                if (game !== undefined) {
-                    this.selectedGame = game;
-                }
-            }
-        });
-        ManagerSettings.getSingleton(GameManager.unsetGame()).then(value => {
-            this.settings = value;
-            this.favourites = value.getContext().global.favouriteGames || [];
-            if (value.getContext().global.defaultGame !== undefined) {
-                if (value.getContext().global.defaultStore !== undefined) {
-                    const game = GameManager.gameList
-                        .find(value1 => value1.internalFolderName === value.getContext().global.defaultGame)!;
-
-                    const platform = game.storePlatformMetadata.find(value1 => value1.storePlatform === value.getContext().global.defaultStore)!;
+        this.settings = await ManagerSettings.getSingleton(GameManager.unsetGame());
+        const globalSettings = this.settings.getContext().global;
+        this.viewMode = globalSettings.gameSelectionViewMode;
```

Comment:
Continuing the previous comment, this should probably default to `GameSelectionViewMode.CARD` in case of `undefined` value.